core/src/main/java/de/jplag/highlightextraction/SubSequenceUtil.java

```diff
@@ -0,0 +1,48 @@
+package de.jplag.highlightextraction;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import de.jplag.TokenType;
+
+/**
+ * Utility class that contains the methods that are used by more than one Strategy.
+ */
+public final class SubSequenceUtil {
+    private SubSequenceUtil() {
+        // private constructor to prevent instantiation
+    }
+
+    /**
+     * Calculates all possible sublists with min minSubSequenceSize length of the matchTokenTypes.
+     * @param matchTokenTypes is a List of the match, in TokenTypes, to create the SubSequences used for the
+     * SubSequenceStrategy and ContainedStrategy. The original list is also included. All contiguous (Sub-)lists, that can
+     * be taken from the matchTokenTypes list that have length greater or equal minSubSequenceSize.
+     * @param minSubSequenceSize is the minimum considered size of the sub-Sequence-List.
+     * @return List of all as considered subSequences.
+     */
+    public static List<List<TokenType>> getSubSequences(List<TokenType> matchTokenTypes, int minSubSequenceSize) {
+        int possibleSubSequenceStartPositions = matchTokenTypes.size() - minSubSequenceSize + 1;
+        int subSequencyCount = possibleSubSequenceStartPositions * (possibleSubSequenceStartPositions + 1) / 2;
+        List<List<TokenType>> subSequences = new ArrayList<>(subSequencyCount);
+        if (matchTokenTypes.size() >= minSubSequenceSize) {
```

Comment:
To decrease complexity, please check for the inverse condition before Line 26 and return `List.of()` immediately if that condition is true.